src/MahApps.Metro.IconPacks.Core/PackIconCursorHelper.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Windows;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.IO;
+
+namespace MahApps.Metro.IconPacks
+{
+    public static class PackIconCursorHelper
+    {
+        /// <summary>
+        /// Get the <see cref="Geometry"/> that will be used to create the cursor
+        /// </summary>
+        /// <param name="pathData"> String data of icon geometry.</param>
+        /// <param name="transformGroup"> Transformation group of the icon (we get it from <see cref="BasePackIconImageExtension.GetPathData(object)"/>).</param>
+        /// <param name="width"> Width of cursor (optional, between 1 and 256, default is 32)</param>
+        /// <param name="height"> Width of cursor (optional, between 1 and 256, default is 32)</param>
+        /// <returns>
+        /// A <see cref="Geometry"/> object that contains the icon to be used in the cursor.
+        /// The <see cref="Geometry"/> has all transformations to be scaled to desired size and aligned to top left corner.
+        /// </returns>
+        public static Geometry GetCursorGeometry(string pathData, TransformGroup transformGroup, double width = 32, double height = 32)
+        {
+            if (width < 1 || width > 256)
+            {
+                throw new ArgumentOutOfRangeException(nameof(width) + " should be between 1 and 256.");
+            }
+            if (height < 1 || height > 256)
+            {
+                throw new ArgumentOutOfRangeException(nameof(height) + " should be between 1 and 256.");
+            }
+
+            Geometry geometry = Geometry.Parse(pathData); // This geometry is frozen and can not be modified
+            geometry = geometry.Clone(); // Create a modifiable copy of geometry.
+
+            // Apply icon pack transformation
+            geometry.Transform = transformGroup;
+
+            // The resulting geometry bounds may not be aligned to (0,0) point. We need to shifted.
+            Rect rect = geometry.Bounds;
+            transformGroup.Children.Add(new TranslateTransform(-rect.X, -rect.Y));
+
+            // Apply the requested size.
+            transformGroup.Children.Add(new ScaleTransform(width / rect.Width, height / rect.Height));
+
+
+            geometry.Freeze();
+            return geometry;
+        }
```

Comment:
I tested your changes on a sample and get this result. It seems like that the scaling is wrong for the image itself.

![2021-12-07_10h03_25](https://user-images.githubusercontent.com/658431/145000718-46681397-d447-4670-94b8-c6d76e9d2a52.png)

Reply:
Sorry for the different scaling factor on X,Y dimensions.
I've made changes to correct that.

Best regards